javascript-bank/bank.js

```diff
@@ -0,0 +1,44 @@
+/* exported Bank */
+
+function Bank() {
+  this.nextAccountNumber = 1;
+  this.accounts = [];
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  if (Number.isInteger(balance) && balance > 0) {
+    var newAccount = new Account(this.nextAccountNumber, holder);
+    newAccount.deposit(balance);
+    this.accounts.push(newAccount);
+    this.nextAccountNumber++;
+    return newAccount.number;
+  } else {
+    return null;
+  }
+};
+
+Bank.prototype.getAccount = function (number) {
+  var counter = 0;
+  for (var i = 0; i < this.accounts.length; i++) {
+    if (this.accounts[i].number === number) {
+      return this.accounts[i];
+    } else {
+      counter++;
+    }
+  }
+  if (counter === this.accounts.length) {
+    return null;
+  }
+};
+
+Bank.prototype.getTotalAssets = function () {
+  var total = 0;
+  for (var i = 0; i < this.accounts.length; i++) {
+    total += this.accounts[i].getBalance();
+  }
+  if (this.accounts.length === 0) {
+    return 0;
+  } else {
+    return total;
+  }
```

Comment:
If the `this.accounts` array has a length of zero, the loop above would never start, since `0` is not less than `0`. As a result, total` would still hold its default value of `0`. You don't need these conditionals, you can just `return total`